puffin/src/frame_data.rs

```diff
@@ -428,6 +466,21 @@ impl FrameData {
         write.write_u8(packed_streams.compression_kind as u8)?;
         write.write_all(&packed_streams.bytes)?;
 
+        let mut to_serialize_scopes = Vec::new();
+
+        if send_all_scopes {
+            for scope in scope_collection.scopes_by_id().iter() {
```

Comment:
perhaps this would work?
```suggestion
to_serialize_scopes.extend(scope_collection.scopes_by_id().values().copied());
```